opm/simulators/wells/MultisegmentWell_impl.hpp

```diff
@@ -1493,6 +1498,10 @@ namespace Opm
             }
             case SURFACE_RATE:
             {
+            if (well_type_ == INJECTOR) {
+                const double target_rate = well_controls_get_current_target(well_controls_);
+                control_eq = getSegmentGTotal(0) - target_rate;
```

Comment:
This is wrong for gas injection.

Reply:
I completely forgot that we discussed this last week. Hopefully, my fix will work. Testing in progress.